### PR TITLE
new utility methods, move some things around and remove deprecated uses

### PR DESCRIPTION
**Pull request by fommil**
preparation for #761 and decoupling with #1437

### CODE REVIEW

**Comment by fommil** [comment]
@harsh FYI

File: util/src/main/scala/org/ensime/util/path.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    def isScala: Boolean = path.toString.toLowerCase.endsWith(".scala")
+    def exists(): Boolean = Files.exists(path)
+    def readBytes(): Array[Byte] = Files.readAllBytes(path)
+    def readString()(implicit cs: Charset): String = new String(readBytes(), cs)
+
+    def /(child: String): Path = path.resolve(child)
+    def canon: Path = ???
+    def isFile: Boolean = path.toFile.isFile
+
+    /*
```
Comment (rorygraves) [inline util/src/main/scala/org/ensime/util/path.scala]: Is this commented code supposed to be in here?

Comment (fommil) [inline util/src/main/scala/org/ensime/util/path.scala]: yeah, it's a reminder about how to use the NIO API, we've to come back to this for the task to remove VFS.

Comment (rorygraves) [inline util/src/main/scala/org/ensime/util/path.scala]: Fair enough

**Comment by rorygraves** [comment]
LGTM (obviously once it actually passes tests ;) )

**Comment by fommil** [comment]
interesting, 

```
info]   java.lang.IllegalArgumentException: Illegal character in opaque part at index 13: jar:file://C:\Program Files\Java\jdk1.7.0\src.zip!\java\lang\String.java
```

**Comment by fommil** [comment]
and the travis environment is a bit weird: 

```
 java.nio.file.FileSystemNotFoundException: /Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/src.zip
```

**Comment by fommil** [comment]
better...

```
ArchiveFile(C:\Program%20Files\Java\jdk1.7.0\src.zip,/) was not equal to ArchiveFile(C:\Program Files\Java\jdk1.7.0\src.zip,/) (EnsimeFileSpec.scala:44)
```

**Pull request merged**